posix/unix.c

```diff
@@ -866,14 +866,23 @@ int unix_shutdown(unsigned socket, int how)
 }
 
 
-/* TODO: support buffer reallocation */
+/* TODO: copy data from old buffer */
 static int unix_bufferSetSize(unixsock_t *s, int sz)
 {
+	void *v;
+
 	if (sz < US_MIN_BUFFER_SIZE || sz > US_MAX_BUFFER_SIZE)
 		return -EINVAL;
 
-	if (s->buffer.data != NULL)
-		return -EINVAL;
+	if (s->buffer.data != NULL) {
+		if ((v = vm_kmalloc(sz)) == NULL)
+			return -ENOMEM;
+
+		proc_lockSet(&s->lock);
+		vm_kfree(s->buffer.data);
```

Comment:
I would be nicer to do `vm_kfree()` outside of the lock (e.g. after new buffer init).